### PR TITLE
update ModifyAsset transaction to pass optional params

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

File: examples/restricted-assets/scripts/1-assetConfig.js [hunk 1]
```diff
@@ -22,21 +22,14 @@ async function run (runtimeEnv, deployer) {
...
-
   /** Update clawback address to escrow **/
   console.log('* Updating asset clawback to escrow *');
   const assetConfigParams = {
     type: types.TransactionType.ModifyAsset,
     sign: types.SignType.SecretKey,
     fromAccount: creator,
     assetID: assetInfo.assetIndex,
-    fields: assetModFields,
+    fields: { clawback: escrowAddress }, // only pass the field you want to update
```
Comment (man-of-code) [inline examples/restricted-assets/scripts/1-assetConfig.js:32]: Instead of removing `assetModFields`, can we change it to the hardcoded object here.

Comment (robert-zaremba) [reply examples/restricted-assets/scripts/1-assetConfig.js:32]: the idea is that we update only the `clawback` address, all rest stays as it's defined in`asa.yaml`

File: packages/algob/src/internal/deployer.ts [hunk 2]
```diff
@@ -96,6 +96,15 @@ class DeployerBasicMode {
     return await this.algoOp.waitForConfirmation(txId);
   }
 
+  /**
+   * Queries blockchain using algodv2 for asset information by index
+   * @param assetIndex asset index
+   * @returns asset info from network
+   */
+  async getAssetByID (assetIndex: number): Promise<algosdk.AssetInfo> {
```
Comment (robert-zaremba) [inline packages/algob/src/internal/deployer.ts:104]: the type should be `number | BigInt`

File: packages/algob/src/lib/algo-operator.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ export interface AlgoOperator {
     txWriter: txWriter,
     scTmplParams?: SCParams) => Promise<SSCInfo>
   waitForConfirmation: (txId: string) => Promise<algosdk.ConfirmedTxInfo>
+  getAssetByID: (assetIndex: number) => Promise<algosdk.AssetInfo>
```
Comment (robert-zaremba) [inline packages/algob/src/lib/algo-operator.ts:47]: ditto

File: packages/algob/src/lib/tx.ts [hunk 4]
```diff
@@ -174,7 +174,25 @@ async function mkTx (
       txIdxMap.set(index, [name, { total: 1, decimals: 1, unitName: "MOCK" }]);
       break;
     }
+    case TransactionType.ModifyAsset: {
+      // fetch asset mutable properties from network and set them (if they are not passed)
+      // before modifying asset
+      const assetInfo = await deployer.getAssetByID(txn.assetID);
+      txn.fields.manager =
+        txn.fields.manager !== "" ? txn.fields.manager ?? assetInfo.params.manager : undefined;
```
Comment (robert-zaremba) [inline packages/algob/src/lib/tx.ts:182]: It's more clear if we put a condition when to update. How about this:
```suggestion
      if (txn.fields.manager === "" ) txn.fields.manager = undefined;
```

File: packages/algob/src/types.ts [hunk 5]
```diff
@@ -493,6 +493,10 @@ export interface Deployer {
    * Queries blockchain for a given transaction and waits until it will be processed. */
   waitForConfirmation: (txId: string) => Promise<algosdk.ConfirmedTxInfo>
 
+  /**
+   * Queries blockchain using algodv2 for asset information by index  */
+  getAssetByID: (assetIndex: number) => Promise<algosdk.AssetInfo>
```
Comment (robert-zaremba) [inline packages/algob/src/types.ts:498]: type should be updated

**Review by robert-zaremba — APPROVED** [review]
pre-approving.

File: packages/algob/src/internal/deployer.ts (hunk 2 above)
Comment (ratik21) [reply packages/algob/src/internal/deployer.ts:104]: yes

File: packages/algob/src/lib/algo-operator.ts (hunk 3 above)
Comment (ratik21) [reply packages/algob/src/lib/algo-operator.ts:47]: updated

File: packages/algob/src/types.ts (hunk 5 above)
Comment (ratik21) [reply packages/algob/src/types.ts:498]: done

File: packages/algob/src/lib/tx.ts (hunk 4 above)
Comment (ratik21) [reply packages/algob/src/lib/tx.ts:182]: yes, thanks for suggestion. Updating.

**Pull request merged**